// @elofun/elo-services/shop/MsgSubscribeResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSubscribeResponse = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
const buffer_1 = require("buffer/");
class MsgSubscribeResponse extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_SUBSCRIBE_RESPONSE, {
            cmd: types_1.UINT8,
            requestId: types_1.UINT16,
            success: types_1.BOOL,
            code: types_1.UINT16,
            msg: types_1.BINARY,
        });
        this.requestId = 0;
        this.cmd = 0;
        this.success = false;
        this.code = 0;
        this.msg = buffer_1.Buffer.from('');
    }
    reset() {
        this.requestId = 0;
        this.cmd = 0;
        this.success = false;
        this.code = 0;
        this.msg = buffer_1.Buffer.from('');
    }
}
exports.MsgSubscribeResponse = MsgSubscribeResponse;
