// @elofun/elo-services/shop/MsgShopSyncItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgShopSyncItems = exports.EShopCurrency = exports.EShopItemTypes = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
var EShopItemTypes;
(function (EShopItemTypes) {
    EShopItemTypes[EShopItemTypes["COIN"] = 0] = "COIN";
    EShopItemTypes[EShopItemTypes["DATA"] = 1] = "DATA";
    EShopItemTypes[EShopItemTypes["VOICE"] = 2] = "VOICE";
    EShopItemTypes[EShopItemTypes["SPIN"] = 3] = "SPIN";
})(EShopItemTypes = exports.EShopItemTypes || (exports.EShopItemTypes = {}));
var EShopCurrency;
(function (EShopCurrency) {
    EShopCurrency[EShopCurrency["COIN"] = 0] = "COIN";
    EShopCurrency[EShopCurrency["CASH"] = 1] = "CASH";
})(EShopCurrency = exports.EShopCurrency || (exports.EShopCurrency = {}));
class MsgShopSyncItems extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_SHOP_SYNC_ITEMS, {
            items: [
                {
                    id: types_1.UINT16,
                    type: types_1.UINT8,
                    price: types_1.FLOAT,
                    currency: types_1.UINT8,
                    unit: types_1.BINARY,
                    amount: types_1.FLOAT,
                    discount: types_1.FLOAT,
                    spin: types_1.UINT8
                },
            ],
        });
        this.items = [];
    }
}
exports.MsgShopSyncItems = MsgShopSyncItems;
