// @elofun/elo-services/notify/ENotifyCodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ENotifyCodes = void 0;
var ENotifyCodes;
(function (ENotifyCodes) {
    ENotifyCodes[ENotifyCodes["E_LOGIN_CONFLICT"] = 1] = "E_LOGIN_CONFLICT";
    ENotifyCodes[ENotifyCodes["E_SYS_MSG"] = 2] = "E_SYS_MSG";
    ENotifyCodes[ENotifyCodes["E_USER_MSG"] = 3] = "E_USER_MSG";
    ENotifyCodes[ENotifyCodes["E_NOT_ENOUGH_BALANCE"] = 4] = "E_NOT_ENOUGH_BALANCE";
    ENotifyCodes[ENotifyCodes["E_GS_INVALID_ROOMID"] = 5] = "E_GS_INVALID_ROOMID";
    ENotifyCodes[ENotifyCodes["E_GS_INVALID_MATCHING"] = 6] = "E_GS_INVALID_MATCHING";
    ENotifyCodes[ENotifyCodes["E_GS_ROOM_NOT_AVAILABLE"] = 7] = "E_GS_ROOM_NOT_AVAILABLE";
    ENotifyCodes[ENotifyCodes["E_GS_NOT_ENOUGH_BALANCE"] = 8] = "E_GS_NOT_ENOUGH_BALANCE";
    ENotifyCodes[ENotifyCodes["E_GS_START_ROOM_FAILED"] = 9] = "E_GS_START_ROOM_FAILED";
    ENotifyCodes[ENotifyCodes["E_GS_LOCAL_ROOM_FULL"] = 10] = "E_GS_LOCAL_ROOM_FULL";
    ENotifyCodes[ENotifyCodes["E_GS_LOCAL_ROOM_CLOSED_BY_HOST"] = 11] = "E_GS_LOCAL_ROOM_CLOSED_BY_HOST";
    ENotifyCodes[ENotifyCodes["E_FORCE_UPDATE"] = 12] = "E_FORCE_UPDATE";
    ENotifyCodes[ENotifyCodes["ESPORT_INVITE"] = 13] = "ESPORT_INVITE";
    ENotifyCodes[ENotifyCodes["ESPORT_JOIN_ROOM"] = 14] = "ESPORT_JOIN_ROOM";
    ENotifyCodes[ENotifyCodes["SPIN_WINNER"] = 15] = "SPIN_WINNER";
})(ENotifyCodes = exports.ENotifyCodes || (exports.ENotifyCodes = {}));
