// @elofun/elo-protobuf/types.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
var ELOTypes;
(function (ELOTypes) {
    ELOTypes[ELOTypes["UINT8"] = 1] = "UINT8";
    ELOTypes[ELOTypes["UINT16"] = 2] = "UINT16";
    ELOTypes[ELOTypes["UINT32"] = 3] = "UINT32";
    ELOTypes[ELOTypes["INT8"] = 4] = "INT8";
    ELOTypes[ELOTypes["INT16"] = 5] = "INT16";
    ELOTypes[ELOTypes["INT32"] = 6] = "INT32";
    ELOTypes[ELOTypes["FLOAT"] = 7] = "FLOAT";
    ELOTypes[ELOTypes["BOOL"] = 8] = "BOOL";
    ELOTypes[ELOTypes["BINARY"] = 9] = "BINARY";
})(ELOTypes = exports.ELOTypes || (exports.ELOTypes = {}));
var UINT8 = ELOTypes.UINT8;
exports.UINT8 = UINT8;
var UINT16 = ELOTypes.UINT16;
exports.UINT16 = UINT16;
var UINT32 = ELOTypes.UINT32;
exports.UINT32 = UINT32;
var INT8 = ELOTypes.INT8;
exports.INT8 = INT8;
var INT16 = ELOTypes.INT16;
exports.INT16 = INT16;
var INT32 = ELOTypes.INT32;
exports.INT32 = INT32;
var FLOAT = ELOTypes.FLOAT;
exports.FLOAT = FLOAT;
var BOOL = ELOTypes.BOOL;
exports.BOOL = BOOL;
var BINARY = ELOTypes.BINARY;
exports.BINARY = BINARY;
exports.ELOSizes = (_a = {},
    _a[UINT8] = 1,
    _a[UINT16] = 2,
    _a[UINT32] = 4,
    _a[INT8] = 1,
    _a[INT16] = 2,
    _a[INT32] = 4,
    _a[FLOAT] = 4,
    _a[BOOL] = 1,
    _a[BINARY] = 1,
    _a);
