// @elofun/elo-services/championship/MsgSyncChampionShip.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncChampionShip = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
const buffer_1 = require("buffer/");
class MsgSyncChampionShip extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_CHAMPIONSHIP_SYNC, {
            startDate: types_1.UINT32,
            endDate: types_1.UINT32,
            isActive: types_1.BOOL,
            prize: types_1.UINT32,
            name: types_1.BINARY,
            unit: types_1.BINARY,
            rank: types_1.UINT32,
            score: types_1.UINT32,
            weekOffset: types_1.UINT8,
            top: [
                {
                    name: types_1.BINARY,
                    avatar: types_1.BINARY,
                    score: types_1.UINT32,
                    rank: types_1.UINT32,
                    userId: types_1.UINT32,
                },
            ],
        });
        this.startDate = 0;
        this.endDate = 0;
        this.isActive = false;
        this.prize = 0;
        this.name = buffer_1.Buffer.from('');
        this.unit = buffer_1.Buffer.from('');
        this.rank = 0;
        this.score = 0;
        this.top = [];
        this.weekOffset = 0;
    }
}
exports.MsgSyncChampionShip = MsgSyncChampionShip;
