// @elofun/elo-services/championship/MsgSyncBigEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncBigEvent = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
const buffer_1 = require("buffer/");
class MsgSyncBigEvent extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_BIG_EVENT_SYNC, {
            name: types_1.BINARY,
            description: types_1.BINARY,
            startTime: types_1.UINT32,
            endTime: types_1.UINT32,
            prizes: [{ rank: types_1.UINT32, amount: types_1.UINT32, unit: types_1.BINARY }],
            feeAmount: types_1.UINT32,
            feeUnit: types_1.BINARY,
            games: [types_1.BINARY],
            top: [
                {
                    name: types_1.BINARY,
                    avatar: types_1.BINARY,
                    score: types_1.UINT32,
                    rank: types_1.UINT32,
                    userId: types_1.UINT32,
                },
            ],
            participants: types_1.UINT32,
            registered: types_1.BOOL,
            rank: types_1.UINT32,
            score: types_1.UINT32,
        });
        this.name = buffer_1.Buffer.from('');
        this.description = buffer_1.Buffer.from('');
        this.startTime = 0;
        this.endTime = 0;
        this.prizes = [];
        this.feeAmount = 0;
        this.feeUnit = buffer_1.Buffer.from('');
        this.games = [];
        this.top = [];
        this.participants = 0;
        this.registered = false;
        this.rank = 0;
        this.score = 0;
    }
}
exports.MsgSyncBigEvent = MsgSyncBigEvent;
