// @elofun/elo-services/gs/MsgClientInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgClientInfo = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const buffer_1 = require("buffer/");
const EMsgTypes_1 = require("../EMsgTypes");
class MsgClientInfo extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_GS_CLIENT_INFO, {
            playerId: types_1.UINT8,
            playerName: types_1.BINARY,
            playerAvatar: types_1.BINARY,
            schema: types_1.BINARY,
            gameState: types_1.BINARY,
            betAmount: types_1.UINT16,
            userId: types_1.UINT32,
        });
        this.playerId = 0;
        this.playerName = buffer_1.Buffer.from('');
        this.playerAvatar = buffer_1.Buffer.from('');
        this.schema = buffer_1.Buffer.from('');
        this.gameState = buffer_1.Buffer.from('');
        this.betAmount = 0;
        this.userId = 0;
    }
}
exports.MsgClientInfo = MsgClientInfo;
