// @elofun/elo-services/profile/MsgSyncProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncProfile = void 0;
const buffer_1 = require("buffer/");
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
class MsgSyncProfile extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_PROFILE_SYNC, {
            userId: types_1.UINT32,
            avatar: types_1.BINARY,
            createdDate: types_1.UINT32,
            credential: types_1.BINARY,
            email: types_1.BINARY,
            lastModified: types_1.UINT32,
            nickName: types_1.BINARY,
            coinBalance: types_1.UINT32,
            spinTicketBalance: types_1.UINT32,
            isBot: types_1.BOOL,
            championShipPoint: types_1.UINT32,
            language: types_1.BINARY,
            isVip: types_1.BOOL,
            isRedeemInviteCode: types_1.BOOL,
        });
        this.reset();
    }
    reset() {
        this.userId = 0;
        this.avatar = buffer_1.Buffer.from('');
        this.createdDate = 0;
        this.credential = buffer_1.Buffer.from('');
        this.email = buffer_1.Buffer.from('');
        this.lastModified = 0;
        this.nickName = buffer_1.Buffer.from('');
        this.coinBalance = 0;
        this.spinTicketBalance = 0;
        this.isBot = false;
        this.championShipPoint = 0;
        this.language = buffer_1.Buffer.from('en');
        this.isVip = false;
        this.isRedeemInviteCode = false;
        return this;
    }
}
exports.MsgSyncProfile = MsgSyncProfile;
