// @elofun/elo-services/profile/MsgSyncActivity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncActivities = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
class MsgSyncActivities extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_SYNC_ACTIVITIES, {
            data: [
                {
                    type: types_1.BINARY,
                    ts: types_1.UINT32,
                    index: types_1.UINT16,
                    content: types_1.BINARY,
                },
            ],
            count: types_1.UINT16,
            offset: types_1.UINT16,
            limit: types_1.UINT8,
        });
        this.data = [];
        this.count = 0;
        this.offset = 0;
        this.limit = 0;
    }
    reset() {
        this.data = [];
        this.count = 0;
        this.offset = 0;
        this.limit = 0;
    }
}
exports.MsgSyncActivities = MsgSyncActivities;
