// @elofun/elo-services/profile/MsgSyncStats.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncStats = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
class MsgSyncStats extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_SYNC_STATS, {
            last_active_dayNo: types_1.UINT32,
            streak_playing_day: types_1.UINT32,
            total_playing_day: types_1.UINT32,
            total_mode_0: types_1.UINT32,
            total_mode_1: types_1.UINT32,
            total_mode_2: types_1.UINT32,
            user_recently_play_mode_0: [
                {
                    opponentId: types_1.UINT32,
                    gameId: types_1.UINT8,
                    utcTime: types_1.BINARY,
                },
            ],
        });
        this.last_active_dayNo = 0;
        this.streak_playing_day = 0;
        this.total_playing_day = 0;
        this.total_mode_0 = 0;
        this.total_mode_1 = 0;
        this.total_mode_2 = 0;
        this.user_recently_play_mode_0 = [];
    }
    reset() {
        this.last_active_dayNo = 0;
        this.streak_playing_day = 0;
        this.total_playing_day = 0;
        this.total_mode_0 = 0;
        this.total_mode_1 = 0;
        this.total_mode_2 = 0;
        this.user_recently_play_mode_0 = [];
        return this;
    }
}
exports.MsgSyncStats = MsgSyncStats;
