// @elofun/elo-services/EApiRequestTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EApiRequestTypes = void 0;
var EApiRequestTypes;
(function (EApiRequestTypes) {
    EApiRequestTypes[EApiRequestTypes["CLAIM_VOUCHER"] = 1] = "CLAIM_VOUCHER";
    EApiRequestTypes[EApiRequestTypes["ESPORT_QRCODE_LOGIN"] = 2] = "ESPORT_QRCODE_LOGIN";
    EApiRequestTypes[EApiRequestTypes["ESPORT_GET_PUBLIC_PROFILE"] = 3] = "ESPORT_GET_PUBLIC_PROFILE";
    EApiRequestTypes[EApiRequestTypes["ESPORT_SEND_INVITE"] = 4] = "ESPORT_SEND_INVITE";
    EApiRequestTypes[EApiRequestTypes["ESPORT_CREATE_ROOM"] = 5] = "ESPORT_CREATE_ROOM";
    EApiRequestTypes[EApiRequestTypes["ESPORT_JOIN_ROOM"] = 6] = "ESPORT_JOIN_ROOM";
    EApiRequestTypes[EApiRequestTypes["ESPORT_START_GAME"] = 7] = "ESPORT_START_GAME";
    EApiRequestTypes[EApiRequestTypes["ESPORT_REJECT"] = 8] = "ESPORT_REJECT";
    EApiRequestTypes[EApiRequestTypes["ADS_REWARD"] = 9] = "ADS_REWARD";
    EApiRequestTypes[EApiRequestTypes["SPIN_GET_WINNER"] = 10] = "SPIN_GET_WINNER";
})(EApiRequestTypes = exports.EApiRequestTypes || (exports.EApiRequestTypes = {}));
