// @elofun/elo-services/EMsgTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EMsgTypes = void 0;
var EMsgTypes;
(function (EMsgTypes) {
    EMsgTypes[EMsgTypes["MSG_CONNECT"] = 0] = "MSG_CONNECT";
    EMsgTypes[EMsgTypes["MSG_DISCONNECTED"] = 1] = "MSG_DISCONNECTED";
    EMsgTypes[EMsgTypes["MSG_AUTH_LOGIN"] = 2] = "MSG_AUTH_LOGIN";
    EMsgTypes[EMsgTypes["MSG_NOTIFY_MSG"] = 3] = "MSG_NOTIFY_MSG";
    EMsgTypes[EMsgTypes["MSG_PROFILE_SYNC"] = 4] = "MSG_PROFILE_SYNC";
    EMsgTypes[EMsgTypes["MSG_MATCHING_REQUEST"] = 5] = "MSG_MATCHING_REQUEST";
    EMsgTypes[EMsgTypes["MSG_MATCHING_START_TOURNAMENT"] = 6] = "MSG_MATCHING_START_TOURNAMENT";
    EMsgTypes[EMsgTypes["MSG_MATCHING_SYNC_TOURNAMENT"] = 7] = "MSG_MATCHING_SYNC_TOURNAMENT";
    EMsgTypes[EMsgTypes["MSG_MATCHING_REQUEST_TOURNAMENT"] = 8] = "MSG_MATCHING_REQUEST_TOURNAMENT";
    EMsgTypes[EMsgTypes["MSG_MATCHING_CREATE_LOCAL_ROOM"] = 9] = "MSG_MATCHING_CREATE_LOCAL_ROOM";
    EMsgTypes[EMsgTypes["MSG_MATCHING_JOIN_LOCAL_ROOM"] = 10] = "MSG_MATCHING_JOIN_LOCAL_ROOM";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM_REQUEST_INFO"] = 11] = "MSG_MATCHING_LOCAL_ROOM_REQUEST_INFO";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM_INFO"] = 12] = "MSG_MATCHING_LOCAL_ROOM_INFO";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM"] = 13] = "MSG_MATCHING_LOCAL_ROOM";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM_SYNC_PLAYERS"] = 14] = "MSG_MATCHING_LOCAL_ROOM_SYNC_PLAYERS";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LEAVE_LOCAL_ROOM"] = 15] = "MSG_MATCHING_LEAVE_LOCAL_ROOM";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM_SELECT_GAME"] = 16] = "MSG_MATCHING_LOCAL_ROOM_SELECT_GAME";
    EMsgTypes[EMsgTypes["MSG_MATCHING_LOCAL_ROOM_READY"] = 17] = "MSG_MATCHING_LOCAL_ROOM_READY";
    EMsgTypes[EMsgTypes["MSG_MATCHING_RESPONSE"] = 18] = "MSG_MATCHING_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_GS_READY"] = 19] = "MSG_GS_READY";
    EMsgTypes[EMsgTypes["MSG_GS_START"] = 20] = "MSG_GS_START";
    EMsgTypes[EMsgTypes["MSG_GS_UPDATE"] = 21] = "MSG_GS_UPDATE";
    EMsgTypes[EMsgTypes["MSG_GS_ENDGAME"] = 22] = "MSG_GS_ENDGAME";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_JOINROOM"] = 23] = "MSG_GS_CLIENT_JOINROOM";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_LEAVEROOM"] = 24] = "MSG_GS_CLIENT_LEAVEROOM";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_INFO"] = 25] = "MSG_GS_CLIENT_INFO";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_READY"] = 26] = "MSG_GS_CLIENT_READY";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_INPUT"] = 27] = "MSG_GS_CLIENT_INPUT";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_CONNECTED"] = 28] = "MSG_GS_CLIENT_CONNECTED";
    EMsgTypes[EMsgTypes["MSG_GS_CLIENT_DISCONNECTED"] = 29] = "MSG_GS_CLIENT_DISCONNECTED";
    EMsgTypes[EMsgTypes["MSG_CHAT"] = 30] = "MSG_CHAT";
    EMsgTypes[EMsgTypes["MSG_SHOP_SYNC_ITEMS"] = 31] = "MSG_SHOP_SYNC_ITEMS";
    EMsgTypes[EMsgTypes["MSG_SHOP_REQUEST_PURCHASE"] = 32] = "MSG_SHOP_REQUEST_PURCHASE";
    EMsgTypes[EMsgTypes["MSG_SHOP_REQUEST_REDEEM"] = 33] = "MSG_SHOP_REQUEST_REDEEM";
    EMsgTypes[EMsgTypes["MSG_SHOP_RESPONSE_PURCHASE"] = 34] = "MSG_SHOP_RESPONSE_PURCHASE";
    EMsgTypes[EMsgTypes["MSG_SHOP_RESPONSE_REDEEM"] = 35] = "MSG_SHOP_RESPONSE_REDEEM";
    EMsgTypes[EMsgTypes["MSG_CHAMPIONSHIP_SYNC"] = 36] = "MSG_CHAMPIONSHIP_SYNC";
    EMsgTypes[EMsgTypes["MSG_LB_LOAD_MORE"] = 37] = "MSG_LB_LOAD_MORE";
    EMsgTypes[EMsgTypes["MSG_LB_ENTRIES"] = 38] = "MSG_LB_ENTRIES";
    EMsgTypes[EMsgTypes["MSG_SYNC_REWARDS"] = 39] = "MSG_SYNC_REWARDS";
    EMsgTypes[EMsgTypes["MSG_CLAIM_REWARD"] = 40] = "MSG_CLAIM_REWARD";
    EMsgTypes[EMsgTypes["MSG_CLAIM_REWARD_RESULT"] = 41] = "MSG_CLAIM_REWARD_RESULT";
    EMsgTypes[EMsgTypes["MSG_SYNC_PLAYTIMES"] = 42] = "MSG_SYNC_PLAYTIMES";
    EMsgTypes[EMsgTypes["MSG_UPDATE_AVATAR"] = 43] = "MSG_UPDATE_AVATAR";
    EMsgTypes[EMsgTypes["MSG_SYNC_ACTIVITIES"] = 44] = "MSG_SYNC_ACTIVITIES";
    EMsgTypes[EMsgTypes["MSG_SYNC_NOTIFICATION"] = 45] = "MSG_SYNC_NOTIFICATION";
    EMsgTypes[EMsgTypes["MSG_SYNC_POINTCUT"] = 46] = "MSG_SYNC_POINTCUT";
    EMsgTypes[EMsgTypes["MSG_UPDATE_POINTCUT"] = 47] = "MSG_UPDATE_POINTCUT";
    EMsgTypes[EMsgTypes["MSG_SYNC_STATS"] = 48] = "MSG_SYNC_STATS";
    EMsgTypes[EMsgTypes["MSG_FB_SHARE_REQUEST"] = 49] = "MSG_FB_SHARE_REQUEST";
    EMsgTypes[EMsgTypes["MSG_FB_SHARE_RESPONSE"] = 50] = "MSG_FB_SHARE_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_FB_SHARE_SUCCESS"] = 51] = "MSG_FB_SHARE_SUCCESS";
    EMsgTypes[EMsgTypes["MSG_FB_SHARE_SYNC"] = 52] = "MSG_FB_SHARE_SYNC";
    EMsgTypes[EMsgTypes["MSG_SYNC_FRIEND"] = 53] = "MSG_SYNC_FRIEND";
    EMsgTypes[EMsgTypes["MSG_FRIEND_REQUEST"] = 54] = "MSG_FRIEND_REQUEST";
    EMsgTypes[EMsgTypes["MSG_FRIEND_ACCEPT"] = 55] = "MSG_FRIEND_ACCEPT";
    EMsgTypes[EMsgTypes["MSG_FRIEND_DECLINE"] = 56] = "MSG_FRIEND_DECLINE";
    EMsgTypes[EMsgTypes["MSG_FRIEND_REMOVE"] = 57] = "MSG_FRIEND_REMOVE";
    EMsgTypes[EMsgTypes["MSG_USER_CARD_REQUEST"] = 58] = "MSG_USER_CARD_REQUEST";
    EMsgTypes[EMsgTypes["MSG_USER_CARD_RESPONSE"] = 59] = "MSG_USER_CARD_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_FRIEND_INVITE_PLAY"] = 60] = "MSG_FRIEND_INVITE_PLAY";
    EMsgTypes[EMsgTypes["MSG_FRIEND_INVITE_ROOM_INFO"] = 61] = "MSG_FRIEND_INVITE_ROOM_INFO";
    EMsgTypes[EMsgTypes["MSG_FRIEND_CODE_REQUEST"] = 62] = "MSG_FRIEND_CODE_REQUEST";
    EMsgTypes[EMsgTypes["MSG_FRIEND_CODE_RESPONSE"] = 63] = "MSG_FRIEND_CODE_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_SPIN_REQUEST"] = 64] = "MSG_SPIN_REQUEST";
    EMsgTypes[EMsgTypes["MSG_SPIN_RESPONSE"] = 65] = "MSG_SPIN_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_SUBSCRIBE_REQUEST"] = 66] = "MSG_SUBSCRIBE_REQUEST";
    EMsgTypes[EMsgTypes["MSG_SUBSCRIBE_RESPONSE"] = 67] = "MSG_SUBSCRIBE_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_API_REQUEST"] = 68] = "MSG_API_REQUEST";
    EMsgTypes[EMsgTypes["MSG_API_RESPONSE"] = 69] = "MSG_API_RESPONSE";
    EMsgTypes[EMsgTypes["MSG_BIG_EVENT_SYNC"] = 70] = "MSG_BIG_EVENT_SYNC";
})(EMsgTypes = exports.EMsgTypes || (exports.EMsgTypes = {}));
