// @elofun/elo-services/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgAPIResponse = exports.MsgAPIRequest = exports.MsgSpinResponse = exports.MsgSpinRequest = exports.MsgUserCardResponse = exports.MsgUserCardRequest = exports.MsgFriendCodeResponse = exports.MsgFriendCodeRequest = exports.MsgFriendInviteRoomInfo = exports.MsgFriendInvitePlayGame = exports.MsgFriendRequestDecline = exports.MsgFriendRequestAccept = exports.MsgFriendRemove = exports.MsgFriendRequest = exports.MsgSyncFriend = exports.MsgFBShareSync = exports.MsgFBShareSuccess = exports.MsgFBShareResponse = exports.MsgFBShareRequest = exports.MsgSyncStats = exports.MsgUpdatePointcut = exports.MsgSyncPointcut = exports.MsgSyncNotification = exports.MsgSyncActivities = exports.MsgSyncPlayTimes = exports.MsgClaimRewardResult = exports.MsgClaimReward = exports.MsgSyncRewards = exports.MsgLBLoadMore = exports.MsgLBEntries = exports.MsgSyncChampionShip = exports.MsgSyncBigEvent = exports.MsgSubscribeResponse = exports.MsgSubscribeRequest = exports.MsgRedeemItemResponse = exports.MsgRedeenItemRequest = exports.MsgPurchaseItemResponse = exports.MsgPurchaseItemRequest = exports.MsgShopSyncItems = exports.MsgChat = exports.MsgClientLeaveRoom = exports.MsgClientJoinRoom = exports.MsgGSEndgame = exports.MsgGSUpdate = exports.MsgGSStart = exports.MsgGSReady = exports.MsgClientReady = exports.MsgClientInput = exports.MsgClientInfo = exports.MsgClientDisconnected = exports.MsgClientConnected = exports.MsgMatchingLocalRoomReady = exports.MsgMatchingLeaveLocalRoom = exports.MsgMatchingLocalRoomSelectGame = exports.MsgMatchingJoinLocalRoom = exports.MsgMatchingLocalRoomInfo = exports.MsgMatchingCreateLocalRoom = exports.MsgMatchingRequestTournament = exports.MsgMatchingSyncTournament = exports.MsgMatchingStartTournament = exports.MsgMatchingResponse = exports.MsgMatchingRequest = exports.MsgUpdateAvatar = exports.MsgSyncProfile = exports.EApiRequestTypes = exports.ENotifyCodes = exports.MsgNotify = exports.ELOMessage = exports.MsgAuthLogin = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
Object.defineProperty(exports, "ELOMessage", { enumerable: true, get: function () { return message_1.ELOMessage; } });
const MsgNotify_1 = require("./notify/MsgNotify");
Object.defineProperty(exports, "MsgNotify", { enumerable: true, get: function () { return MsgNotify_1.MsgNotify; } });
const MsgAuthLogin_1 = require("./auth/MsgAuthLogin");
Object.defineProperty(exports, "MsgAuthLogin", { enumerable: true, get: function () { return MsgAuthLogin_1.MsgAuthLogin; } });
const ENotifyCodes_1 = require("./notify/ENotifyCodes");
Object.defineProperty(exports, "ENotifyCodes", { enumerable: true, get: function () { return ENotifyCodes_1.ENotifyCodes; } });
const MsgSyncProfile_1 = require("./profile/MsgSyncProfile");
Object.defineProperty(exports, "MsgSyncProfile", { enumerable: true, get: function () { return MsgSyncProfile_1.MsgSyncProfile; } });
const MsgMatchingRequest_1 = require("./matching/MsgMatchingRequest");
Object.defineProperty(exports, "MsgMatchingRequest", { enumerable: true, get: function () { return MsgMatchingRequest_1.MsgMatchingRequest; } });
const MsgMatchingResponse_1 = require("./matching/MsgMatchingResponse");
Object.defineProperty(exports, "MsgMatchingResponse", { enumerable: true, get: function () { return MsgMatchingResponse_1.MsgMatchingResponse; } });
const MsgClientConnected_1 = require("./gs/MsgClientConnected");
Object.defineProperty(exports, "MsgClientConnected", { enumerable: true, get: function () { return MsgClientConnected_1.MsgClientConnected; } });
const MsgClientDisconnected_1 = require("./gs/MsgClientDisconnected");
Object.defineProperty(exports, "MsgClientDisconnected", { enumerable: true, get: function () { return MsgClientDisconnected_1.MsgClientDisconnected; } });
const MsgClientInfo_1 = require("./gs/MsgClientInfo");
Object.defineProperty(exports, "MsgClientInfo", { enumerable: true, get: function () { return MsgClientInfo_1.MsgClientInfo; } });
const MsgClientInput_1 = require("./gs/MsgClientInput");
Object.defineProperty(exports, "MsgClientInput", { enumerable: true, get: function () { return MsgClientInput_1.MsgClientInput; } });
const MsgClientReady_1 = require("./gs/MsgClientReady");
Object.defineProperty(exports, "MsgClientReady", { enumerable: true, get: function () { return MsgClientReady_1.MsgClientReady; } });
const MsgGSStart_1 = require("./gs/MsgGSStart");
Object.defineProperty(exports, "MsgGSStart", { enumerable: true, get: function () { return MsgGSStart_1.MsgGSStart; } });
const MsgGSUpdate_1 = require("./gs/MsgGSUpdate");
Object.defineProperty(exports, "MsgGSUpdate", { enumerable: true, get: function () { return MsgGSUpdate_1.MsgGSUpdate; } });
const MsgGSEndgame_1 = require("./gs/MsgGSEndgame");
Object.defineProperty(exports, "MsgGSEndgame", { enumerable: true, get: function () { return MsgGSEndgame_1.MsgGSEndgame; } });
const MsgClientJoinRoom_1 = require("./gs/MsgClientJoinRoom");
Object.defineProperty(exports, "MsgClientJoinRoom", { enumerable: true, get: function () { return MsgClientJoinRoom_1.MsgClientJoinRoom; } });
const MsgClientLeaveRoom_1 = require("./gs/MsgClientLeaveRoom");
Object.defineProperty(exports, "MsgClientLeaveRoom", { enumerable: true, get: function () { return MsgClientLeaveRoom_1.MsgClientLeaveRoom; } });
const MsgGSReady_1 = require("./gs/MsgGSReady");
Object.defineProperty(exports, "MsgGSReady", { enumerable: true, get: function () { return MsgGSReady_1.MsgGSReady; } });
const MsgMatchingStartTournament_1 = require("./matching/MsgMatchingStartTournament");
Object.defineProperty(exports, "MsgMatchingStartTournament", { enumerable: true, get: function () { return MsgMatchingStartTournament_1.MsgMatchingStartTournament; } });
const MsgMatchingSyncTournament_1 = require("./matching/MsgMatchingSyncTournament");
Object.defineProperty(exports, "MsgMatchingSyncTournament", { enumerable: true, get: function () { return MsgMatchingSyncTournament_1.MsgMatchingSyncTournament; } });
const MsgMatchingRequestTournament_1 = require("./matching/MsgMatchingRequestTournament");
Object.defineProperty(exports, "MsgMatchingRequestTournament", { enumerable: true, get: function () { return MsgMatchingRequestTournament_1.MsgMatchingRequestTournament; } });
const MsgMatchingCreateLocalRoom_1 = require("./matching/MsgMatchingCreateLocalRoom");
Object.defineProperty(exports, "MsgMatchingCreateLocalRoom", { enumerable: true, get: function () { return MsgMatchingCreateLocalRoom_1.MsgMatchingCreateLocalRoom; } });
const MsgChat_1 = require("./profile/MsgChat");
Object.defineProperty(exports, "MsgChat", { enumerable: true, get: function () { return MsgChat_1.MsgChat; } });
const MsgShopSyncItems_1 = require("./shop/MsgShopSyncItems");
Object.defineProperty(exports, "MsgShopSyncItems", { enumerable: true, get: function () { return MsgShopSyncItems_1.MsgShopSyncItems; } });
const MsgPurchaseItemRequest_1 = require("./shop/MsgPurchaseItemRequest");
Object.defineProperty(exports, "MsgPurchaseItemRequest", { enumerable: true, get: function () { return MsgPurchaseItemRequest_1.MsgPurchaseItemRequest; } });
const MsgPurchaseItemResponse_1 = require("./shop/MsgPurchaseItemResponse");
Object.defineProperty(exports, "MsgPurchaseItemResponse", { enumerable: true, get: function () { return MsgPurchaseItemResponse_1.MsgPurchaseItemResponse; } });
const MsgRedeenItemRequest_1 = require("./shop/MsgRedeenItemRequest");
Object.defineProperty(exports, "MsgRedeenItemRequest", { enumerable: true, get: function () { return MsgRedeenItemRequest_1.MsgRedeenItemRequest; } });
const MsgRedeemItemResponse_1 = require("./shop/MsgRedeemItemResponse");
Object.defineProperty(exports, "MsgRedeemItemResponse", { enumerable: true, get: function () { return MsgRedeemItemResponse_1.MsgRedeemItemResponse; } });
const MsgSyncBigEvent_1 = require("./championship/MsgSyncBigEvent");
Object.defineProperty(exports, "MsgSyncBigEvent", { enumerable: true, get: function () { return MsgSyncBigEvent_1.MsgSyncBigEvent; } });
const MsgSyncChampionShip_1 = require("./championship/MsgSyncChampionShip");
Object.defineProperty(exports, "MsgSyncChampionShip", { enumerable: true, get: function () { return MsgSyncChampionShip_1.MsgSyncChampionShip; } });
const MsgSyncReward_1 = require("./rewards/MsgSyncReward");
Object.defineProperty(exports, "MsgSyncRewards", { enumerable: true, get: function () { return MsgSyncReward_1.MsgSyncRewards; } });
const MsgSyncPlayTimes_1 = require("./profile/MsgSyncPlayTimes");
Object.defineProperty(exports, "MsgSyncPlayTimes", { enumerable: true, get: function () { return MsgSyncPlayTimes_1.MsgSyncPlayTimes; } });
const MsgUpdateAvatar_1 = require("./profile/MsgUpdateAvatar");
Object.defineProperty(exports, "MsgUpdateAvatar", { enumerable: true, get: function () { return MsgUpdateAvatar_1.MsgUpdateAvatar; } });
const MsgMatchingLocalRoomInfo_1 = require("./matching/MsgMatchingLocalRoomInfo");
Object.defineProperty(exports, "MsgMatchingLocalRoomInfo", { enumerable: true, get: function () { return MsgMatchingLocalRoomInfo_1.MsgMatchingLocalRoomInfo; } });
const MsgMatchingJoinLocalRoom_1 = require("./matching/MsgMatchingJoinLocalRoom");
Object.defineProperty(exports, "MsgMatchingJoinLocalRoom", { enumerable: true, get: function () { return MsgMatchingJoinLocalRoom_1.MsgMatchingJoinLocalRoom; } });
const MsgMatchingLocalRoomSelectGame_1 = require("./matching/MsgMatchingLocalRoomSelectGame");
Object.defineProperty(exports, "MsgMatchingLocalRoomSelectGame", { enumerable: true, get: function () { return MsgMatchingLocalRoomSelectGame_1.MsgMatchingLocalRoomSelectGame; } });
const MsgMatchingLeaveLocalRoom_1 = require("./matching/MsgMatchingLeaveLocalRoom");
Object.defineProperty(exports, "MsgMatchingLeaveLocalRoom", { enumerable: true, get: function () { return MsgMatchingLeaveLocalRoom_1.MsgMatchingLeaveLocalRoom; } });
const MsgMatchingLocalRoomReady_1 = require("./matching/MsgMatchingLocalRoomReady");
Object.defineProperty(exports, "MsgMatchingLocalRoomReady", { enumerable: true, get: function () { return MsgMatchingLocalRoomReady_1.MsgMatchingLocalRoomReady; } });
const MsgClaimReward_1 = require("./rewards/MsgClaimReward");
Object.defineProperty(exports, "MsgClaimReward", { enumerable: true, get: function () { return MsgClaimReward_1.MsgClaimReward; } });
const MsgClaimRewardResult_1 = require("./rewards/MsgClaimRewardResult");
Object.defineProperty(exports, "MsgClaimRewardResult", { enumerable: true, get: function () { return MsgClaimRewardResult_1.MsgClaimRewardResult; } });
const MsgSyncActivity_1 = require("./profile/MsgSyncActivity");
Object.defineProperty(exports, "MsgSyncActivities", { enumerable: true, get: function () { return MsgSyncActivity_1.MsgSyncActivities; } });
const MsgSyncNotification_1 = require("./profile/MsgSyncNotification");
Object.defineProperty(exports, "MsgSyncNotification", { enumerable: true, get: function () { return MsgSyncNotification_1.MsgSyncNotification; } });
const MsgSyncPointcut_1 = require("./profile/MsgSyncPointcut");
Object.defineProperty(exports, "MsgSyncPointcut", { enumerable: true, get: function () { return MsgSyncPointcut_1.MsgSyncPointcut; } });
const MsgUpdatePointcut_1 = require("./profile/MsgUpdatePointcut");
Object.defineProperty(exports, "MsgUpdatePointcut", { enumerable: true, get: function () { return MsgUpdatePointcut_1.MsgUpdatePointcut; } });
const MsgSyncStats_1 = require("./profile/MsgSyncStats");
Object.defineProperty(exports, "MsgSyncStats", { enumerable: true, get: function () { return MsgSyncStats_1.MsgSyncStats; } });
const MsgFBShareRequest_1 = require("./share/MsgFBShareRequest");
Object.defineProperty(exports, "MsgFBShareRequest", { enumerable: true, get: function () { return MsgFBShareRequest_1.MsgFBShareRequest; } });
const MsgFBShareResponse_1 = require("./share/MsgFBShareResponse");
Object.defineProperty(exports, "MsgFBShareResponse", { enumerable: true, get: function () { return MsgFBShareResponse_1.MsgFBShareResponse; } });
const MsgFBShareSuccess_1 = require("./share/MsgFBShareSuccess");
Object.defineProperty(exports, "MsgFBShareSuccess", { enumerable: true, get: function () { return MsgFBShareSuccess_1.MsgFBShareSuccess; } });
const MsgFBShareSync_1 = require("./share/MsgFBShareSync");
Object.defineProperty(exports, "MsgFBShareSync", { enumerable: true, get: function () { return MsgFBShareSync_1.MsgFBShareSync; } });
const MsgSyncFriend_1 = require("./friend/MsgSyncFriend");
Object.defineProperty(exports, "MsgSyncFriend", { enumerable: true, get: function () { return MsgSyncFriend_1.MsgSyncFriend; } });
const MsgFriendRequest_1 = require("./friend/MsgFriendRequest");
Object.defineProperty(exports, "MsgFriendRequest", { enumerable: true, get: function () { return MsgFriendRequest_1.MsgFriendRequest; } });
const MsgFriendRemove_1 = require("./friend/MsgFriendRemove");
Object.defineProperty(exports, "MsgFriendRemove", { enumerable: true, get: function () { return MsgFriendRemove_1.MsgFriendRemove; } });
const MsgFriendRequestAccept_1 = require("./friend/MsgFriendRequestAccept");
Object.defineProperty(exports, "MsgFriendRequestAccept", { enumerable: true, get: function () { return MsgFriendRequestAccept_1.MsgFriendRequestAccept; } });
const MsgFriendRequestDecline_1 = require("./friend/MsgFriendRequestDecline");
Object.defineProperty(exports, "MsgFriendRequestDecline", { enumerable: true, get: function () { return MsgFriendRequestDecline_1.MsgFriendRequestDecline; } });
const MsgFriendInvitePlayGame_1 = require("./friend/MsgFriendInvitePlayGame");
Object.defineProperty(exports, "MsgFriendInvitePlayGame", { enumerable: true, get: function () { return MsgFriendInvitePlayGame_1.MsgFriendInvitePlayGame; } });
const MsgFriendInviteRoomInfo_1 = require("./friend/MsgFriendInviteRoomInfo");
Object.defineProperty(exports, "MsgFriendInviteRoomInfo", { enumerable: true, get: function () { return MsgFriendInviteRoomInfo_1.MsgFriendInviteRoomInfo; } });
const MsgFriendCodeRequest_1 = require("./friend/MsgFriendCodeRequest");
Object.defineProperty(exports, "MsgFriendCodeRequest", { enumerable: true, get: function () { return MsgFriendCodeRequest_1.MsgFriendCodeRequest; } });
const MsgFriendCodeResponse_1 = require("./friend/MsgFriendCodeResponse");
Object.defineProperty(exports, "MsgFriendCodeResponse", { enumerable: true, get: function () { return MsgFriendCodeResponse_1.MsgFriendCodeResponse; } });
const MsgUserCardRequest_1 = require("./profile/MsgUserCardRequest");
Object.defineProperty(exports, "MsgUserCardRequest", { enumerable: true, get: function () { return MsgUserCardRequest_1.MsgUserCardRequest; } });
const MsgUserCardResponse_1 = require("./profile/MsgUserCardResponse");
Object.defineProperty(exports, "MsgUserCardResponse", { enumerable: true, get: function () { return MsgUserCardResponse_1.MsgUserCardResponse; } });
const MsgLBLoadMore_1 = require("./championship/MsgLBLoadMore");
Object.defineProperty(exports, "MsgLBLoadMore", { enumerable: true, get: function () { return MsgLBLoadMore_1.MsgLBLoadMore; } });
const MsgLBEntries_1 = require("./championship/MsgLBEntries");
Object.defineProperty(exports, "MsgLBEntries", { enumerable: true, get: function () { return MsgLBEntries_1.MsgLBEntries; } });
const MsgSpinRequest_1 = require("./spin/MsgSpinRequest");
Object.defineProperty(exports, "MsgSpinRequest", { enumerable: true, get: function () { return MsgSpinRequest_1.MsgSpinRequest; } });
const MsgSpinResponse_1 = require("./spin/MsgSpinResponse");
Object.defineProperty(exports, "MsgSpinResponse", { enumerable: true, get: function () { return MsgSpinResponse_1.MsgSpinResponse; } });
const MsgSubscribeRequest_1 = require("./shop/MsgSubscribeRequest");
Object.defineProperty(exports, "MsgSubscribeRequest", { enumerable: true, get: function () { return MsgSubscribeRequest_1.MsgSubscribeRequest; } });
const MsgSubscribeResponse_1 = require("./shop/MsgSubscribeResponse");
Object.defineProperty(exports, "MsgSubscribeResponse", { enumerable: true, get: function () { return MsgSubscribeResponse_1.MsgSubscribeResponse; } });
const MsgAPIRequest_1 = require("./request/MsgAPIRequest");
Object.defineProperty(exports, "MsgAPIRequest", { enumerable: true, get: function () { return MsgAPIRequest_1.MsgAPIRequest; } });
const MsgAPIResponse_1 = require("./request/MsgAPIResponse");
Object.defineProperty(exports, "MsgAPIResponse", { enumerable: true, get: function () { return MsgAPIResponse_1.MsgAPIResponse; } });
const EApiRequestTypes_1 = require("./EApiRequestTypes");
Object.defineProperty(exports, "EApiRequestTypes", { enumerable: true, get: function () { return EApiRequestTypes_1.EApiRequestTypes; } });
message_1.ELOMessage.register(new MsgAuthLogin_1.MsgAuthLogin());
message_1.ELOMessage.register(new MsgSyncProfile_1.MsgSyncProfile());
message_1.ELOMessage.register(new MsgUpdateAvatar_1.MsgUpdateAvatar());
// notify
message_1.ELOMessage.register(new MsgNotify_1.MsgNotify());
// Matching
message_1.ELOMessage.register(new MsgMatchingRequest_1.MsgMatchingRequest());
message_1.ELOMessage.register(new MsgMatchingResponse_1.MsgMatchingResponse());
message_1.ELOMessage.register(new MsgMatchingStartTournament_1.MsgMatchingStartTournament());
message_1.ELOMessage.register(new MsgMatchingSyncTournament_1.MsgMatchingSyncTournament());
message_1.ELOMessage.register(new MsgMatchingRequestTournament_1.MsgMatchingRequestTournament());
message_1.ELOMessage.register(new MsgMatchingCreateLocalRoom_1.MsgMatchingCreateLocalRoom());
message_1.ELOMessage.register(new MsgMatchingLocalRoomInfo_1.MsgMatchingLocalRoomInfo());
message_1.ELOMessage.register(new MsgMatchingJoinLocalRoom_1.MsgMatchingJoinLocalRoom());
message_1.ELOMessage.register(new MsgMatchingLocalRoomSelectGame_1.MsgMatchingLocalRoomSelectGame());
message_1.ELOMessage.register(new MsgMatchingLeaveLocalRoom_1.MsgMatchingLeaveLocalRoom());
message_1.ELOMessage.register(new MsgMatchingLocalRoomReady_1.MsgMatchingLocalRoomReady());
// GS
message_1.ELOMessage.register(new MsgClientConnected_1.MsgClientConnected());
message_1.ELOMessage.register(new MsgClientDisconnected_1.MsgClientDisconnected());
message_1.ELOMessage.register(new MsgClientInfo_1.MsgClientInfo());
message_1.ELOMessage.register(new MsgClientInput_1.MsgClientInput());
message_1.ELOMessage.register(new MsgClientReady_1.MsgClientReady());
message_1.ELOMessage.register(new MsgGSStart_1.MsgGSStart());
message_1.ELOMessage.register(new MsgGSUpdate_1.MsgGSUpdate());
message_1.ELOMessage.register(new MsgGSEndgame_1.MsgGSEndgame());
message_1.ELOMessage.register(new MsgClientJoinRoom_1.MsgClientJoinRoom());
message_1.ELOMessage.register(new MsgClientLeaveRoom_1.MsgClientLeaveRoom());
message_1.ELOMessage.register(new MsgGSReady_1.MsgGSReady());
//chat
message_1.ELOMessage.register(new MsgChat_1.MsgChat());
//shop
message_1.ELOMessage.register(new MsgShopSyncItems_1.MsgShopSyncItems());
message_1.ELOMessage.register(new MsgPurchaseItemRequest_1.MsgPurchaseItemRequest());
message_1.ELOMessage.register(new MsgPurchaseItemResponse_1.MsgPurchaseItemResponse());
message_1.ELOMessage.register(new MsgRedeenItemRequest_1.MsgRedeenItemRequest());
message_1.ELOMessage.register(new MsgRedeemItemResponse_1.MsgRedeemItemResponse());
message_1.ELOMessage.register(new MsgSubscribeRequest_1.MsgSubscribeRequest());
message_1.ELOMessage.register(new MsgSubscribeResponse_1.MsgSubscribeResponse());
// big event
message_1.ELOMessage.register(new MsgSyncBigEvent_1.MsgSyncBigEvent());
// champion ship
message_1.ELOMessage.register(new MsgSyncChampionShip_1.MsgSyncChampionShip());
message_1.ELOMessage.register(new MsgLBLoadMore_1.MsgLBLoadMore());
message_1.ELOMessage.register(new MsgLBEntries_1.MsgLBEntries());
//reward
message_1.ELOMessage.register(new MsgSyncReward_1.MsgSyncRewards());
message_1.ELOMessage.register(new MsgClaimReward_1.MsgClaimReward());
message_1.ELOMessage.register(new MsgClaimRewardResult_1.MsgClaimRewardResult());
message_1.ELOMessage.register(new MsgSyncPlayTimes_1.MsgSyncPlayTimes());
message_1.ELOMessage.register(new MsgSyncActivity_1.MsgSyncActivities());
message_1.ELOMessage.register(new MsgSyncNotification_1.MsgSyncNotification());
//pointcut
message_1.ELOMessage.register(new MsgSyncPointcut_1.MsgSyncPointcut());
message_1.ELOMessage.register(new MsgUpdatePointcut_1.MsgUpdatePointcut());
// stats
message_1.ELOMessage.register(new MsgSyncStats_1.MsgSyncStats());
// share
message_1.ELOMessage.register(new MsgFBShareRequest_1.MsgFBShareRequest());
message_1.ELOMessage.register(new MsgFBShareResponse_1.MsgFBShareResponse());
message_1.ELOMessage.register(new MsgFBShareSuccess_1.MsgFBShareSuccess());
message_1.ELOMessage.register(new MsgFBShareSync_1.MsgFBShareSync());
// friend
message_1.ELOMessage.register(new MsgSyncFriend_1.MsgSyncFriend());
message_1.ELOMessage.register(new MsgFriendRequest_1.MsgFriendRequest());
message_1.ELOMessage.register(new MsgFriendRemove_1.MsgFriendRemove());
message_1.ELOMessage.register(new MsgFriendRequestAccept_1.MsgFriendRequestAccept());
message_1.ELOMessage.register(new MsgFriendRequestDecline_1.MsgFriendRequestDecline());
message_1.ELOMessage.register(new MsgFriendInvitePlayGame_1.MsgFriendInvitePlayGame());
message_1.ELOMessage.register(new MsgFriendInviteRoomInfo_1.MsgFriendInviteRoomInfo());
message_1.ELOMessage.register(new MsgFriendCodeRequest_1.MsgFriendCodeRequest());
message_1.ELOMessage.register(new MsgFriendCodeResponse_1.MsgFriendCodeResponse());
// user card
message_1.ELOMessage.register(new MsgUserCardRequest_1.MsgUserCardRequest());
message_1.ELOMessage.register(new MsgUserCardResponse_1.MsgUserCardResponse());
// spin
message_1.ELOMessage.register(new MsgSpinRequest_1.MsgSpinRequest());
message_1.ELOMessage.register(new MsgSpinResponse_1.MsgSpinResponse());
// api request
message_1.ELOMessage.register(new MsgAPIRequest_1.MsgAPIRequest());
message_1.ELOMessage.register(new MsgAPIResponse_1.MsgAPIResponse());
