// @elofun/vds-gami-2022-wsgw/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.connectWsgw = void 0;
var elo_protobuf_1 = require("@elofun/elo-protobuf");
var types_1 = require("@elofun/elo-protobuf/types");
var buffer_1 = require("buffer/");
var wswgDict = {};
var connectWsgw = function (ep) { return __awaiter(void 0, void 0, void 0, function () {
    var param;
    return __generator(this, function (_a) {
        if (window.URLSearchParams) {
            param = new URLSearchParams(window.location.search);
            if (param && param.get('host')) {
                ep = param.get('host') || ep;
            }
        }
        if (!wswgDict[ep]) {
            wswgDict[ep] = new Promise(function (resolve) {
                var ctx = {
                    ep: ep,
                    ts: Date.now(),
                    ws: new WebSocket(ep),
                };
                ctx.ws.binaryType = 'arraybuffer';
                ctx.ws.onopen = function () {
                    ctx.ws.send((0, elo_protobuf_1.eloPack)({
                        u: types_1.BINARY,
                        t: types_1.BINARY,
                    }, {
                        u: buffer_1.Buffer.from(window.navigator.userAgent),
                        t: buffer_1.Buffer.from(window.location.href),
                    }));
                    ctx.ws.onmessage = function (e) {
                        var _a;
                        ctx.ws.onmessage = null;
                        setTimeout(Function("\"use strict\";var ctx=arguments[0];".concat(buffer_1.Buffer.concat((0, elo_protobuf_1.eloUnpack)((_a = {}, _a[ctx.ts] = [types_1.BINARY], _a), buffer_1.Buffer.from(e.data))[ctx.ts]), ";arguments[1](ctx.wsgw);")), 0, ctx, resolve);
                    };
                };
            });
        }
        return [2 /*return*/, wswgDict[ep]];
    });
}); };
exports.connectWsgw = connectWsgw;
