// @elofun/elo-services/matching/MsgMatchingRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgMatchingRequest = exports.EMatchingType = void 0;
const buffer_1 = require("buffer/");
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
var EMatchingType;
(function (EMatchingType) {
    EMatchingType[EMatchingType["PVP"] = 0] = "PVP";
    EMatchingType[EMatchingType["TOURNAMENT"] = 1] = "TOURNAMENT";
    EMatchingType[EMatchingType["LOCAL"] = 2] = "LOCAL";
    EMatchingType[EMatchingType["ESPORT"] = 3] = "ESPORT";
})(EMatchingType = exports.EMatchingType || (exports.EMatchingType = {}));
class MsgMatchingRequest extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_MATCHING_REQUEST, {
            bet: types_1.UINT16,
            gameId: types_1.UINT8,
            serverHash: types_1.BINARY,
            bot: types_1.BINARY,
            nonce: types_1.UINT16,
            type: types_1.UINT8,
            capacity: types_1.UINT8,
            reserveCode: types_1.BINARY,
        });
        this.bet = 0;
        this.gameId = 0;
        this.serverHash = buffer_1.Buffer.from('');
        this.bot = buffer_1.Buffer.from('');
        this.nonce = 0;
        this.type = 0;
        this.capacity = 0;
        this.reserveCode = buffer_1.Buffer.from('');
    }
}
exports.MsgMatchingRequest = MsgMatchingRequest;
