// @elofun/elo-services/friend/MsgSyncFriend.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgSyncFriend = void 0;
const message_1 = require("@elofun/elo-protobuf/message");
const types_1 = require("@elofun/elo-protobuf/types");
const EMsgTypes_1 = require("../EMsgTypes");
class MsgSyncFriend extends message_1.ELOMessage {
    constructor() {
        super(EMsgTypes_1.EMsgTypes.MSG_SYNC_FRIEND, {
            friends: [
                {
                    id: types_1.UINT32,
                },
            ],
            requests: [
                {
                    reqId: types_1.UINT16,
                    id: types_1.UINT32,
                    mutual: types_1.UINT8,
                },
            ],
            pendings: [types_1.UINT32],
        });
        this.friends = [];
        this.requests = [];
        this.pendings = [];
    }
    reset() {
        this.friends = [];
        this.requests = [];
        this.pendings = [];
        return this;
    }
}
exports.MsgSyncFriend = MsgSyncFriend;
